plugin/src/main/java/org/opensearch/sql/plugin/transport/PPLQueryHelper.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.plugin.transport;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.action.ActionListener;
+import org.opensearch.commons.sql.action.TransportQueryRequest;
+import org.opensearch.commons.sql.action.TransportQueryResponse;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.sql.common.antlr.SyntaxCheckException;
+import org.opensearch.sql.common.response.ResponseListener;
+import org.opensearch.sql.exception.ExpressionEvaluationException;
+import org.opensearch.sql.exception.QueryEngineException;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.legacy.metrics.MetricName;
+import org.opensearch.sql.legacy.metrics.Metrics;
+import org.opensearch.sql.ppl.PPLService;
+import org.opensearch.sql.ppl.domain.PPLQueryRequest;
+import org.opensearch.sql.protocol.response.QueryResult;
+import org.opensearch.sql.protocol.response.format.CsvResponseFormatter;
+import org.opensearch.sql.protocol.response.format.Format;
+import org.opensearch.sql.protocol.response.format.RawResponseFormatter;
+import org.opensearch.sql.protocol.response.format.ResponseFormatter;
+import org.opensearch.sql.protocol.response.format.SimpleJsonResponseFormatter;
+import org.opensearch.sql.protocol.response.format.VisualizationResponseFormatter;
+
+import static org.opensearch.rest.RestStatus.BAD_REQUEST;
+import static org.opensearch.rest.RestStatus.SERVICE_UNAVAILABLE;
+import static org.opensearch.sql.protocol.response.format.JsonResponseFormatter.Style.PRETTY;
+
+public class PPLQueryHelper {
+    private static PPLService pplService;
+    private static PPLQueryHelper INSTANCE;
+
+    public static synchronized PPLQueryHelper getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new PPLQueryHelper();
+        }
+        return INSTANCE;
+    }
```

Comment:
also set constructor to private for singleton?

Reply:
good pointed, missed that